test/test_cuda.py

```diff
@@ -3280,5 +3280,200 @@ class TestNamedTupleInput_1(NamedTuple):
             self.assertEqual(expected_a, x.a)
             self.assertEqual(expected_b, x.b)
 
+    def test_graph_capture_simple(self):
+        s1 = torch.cuda.Stream()
+
+        with torch.cuda.stream(s1):
+            a = torch.zeros((1000,), device="cuda")
+            a += 1
+            g = torch.cuda.Graph()
+            g.capture_begin()
```

Comment:
you sure you don't want a context manager for this?

Reply:
The manager could also take care of getting onto a non-default stream

Reply:
I think eventually a context manager is the right exposure, but right now, this won't be documented, and exists to serve experiments.  In experiments, I don't want to exit gracefully from graphed regions if capture fails.  I want all hell to break loose so we see it.

Reply:
I'm hacking on some local UXs that take care of non-default stream.  The goal of this PR is to introduce a very lightweight capture exposure to enable UX hacking.